vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -345,14 +345,152 @@ testCustomCATimerNotStarted() {
+  echo "$test: Checking general format of $settings_file"
+  grep -E -v -n -H '^\s*(#|$)' $settings_file | grep -E -v '^.+:.+:\s*[A-Z_]+\s+[^[:cntrl:]]+$' >>/dev/stderr
+  if [ $? -eq 0 ]; then
+    err $test "File $settings_file is not valid; see above for bad lines."
+  fi
+
+  # Look for the settings we specifically set in <repo-root>/parts/linux/cloud-init/artifacts/cis.sh
+  # and ensure they're set to the values we expect.
+  echo "$test: Checking specific settings in $settings_file"
+  testSetting $test $settings_file ' ' PASS_MAX_DAYS 90
+  testSetting $test $settings_file ' ' PASS_MIN_DAYS 7
+
+  echo "$test:Finish"
+}
+
+# Ensures that /etc/default/useradd is valid. This is a best-effort test, as we aren't going to
+# re-implement everything that uses this file.
+testUserAdd() {
+  test="testUserAdd"
+  local settings_file=/etc/default/useradd
+  echo "$test:Start"
+
+  # If the file doesn't exist, everything is broken.
+  echo "$test: Checking existence of $settings_file"
+  if [ ! -f $settings_file ]; then
+    err $test "File $settings_file not found"
+    return
+  fi
+
+  # Do a rough content validation. The man page https://www.man7.org/linux/man-pages/man8/useradd.8.html
+  # doesn't really state the format of the file, but experimentation and examples shows that the file
+  # is a series of lines with the following properties:
+  #   - A line is a comment or a setting.
+  #   - A comment starts with optional whitespace, then a '#' and continues to the end of the line.
+  #   - A setting is a name, followed by an '=', followed by a value. No whitespace is allowed
+  #     before, between, or after these.
+  #   - The name is all-caps and includes letters and '_'.
+  #   - The setting values are generally numbers and letters, which includes paths. Being
+  #     paths, there's a pathological case wehre a setting would have spaces and other escaped
+  #     characters, so we check for printable characters -- meaning non-control characters. The
+  #     value can also be empty.
+  #
+  # The first grep filters out lines that are either comments or empty. The second grep filters
+  # out lines that match the above format, after accounting for the file name and line number
+  # in the output. If there are any lines left, the file is invalid.
+  # We print the bad lines to stderr so that they're visible in the test output.
+  # Grep returns 0 if it finds a match, which would mean the file is invalid.
+  echo "$test: Checking general format of $settings_file"
+  grep -E -v -n -H '^\s*(#|$)' $settings_file | grep -E -v '^.+:.+:[A-Z_]+=[^[:cntrl:]]*$' >>/dev/stderr
+  if [ $? -eq 0 ]; then
+    err $test "File $settings_file is not valid; see above for bad lines."
   fi
+
+  # Look for the settings we specifically set in <repo-root>/parts/linux/cloud-init/artifacts/cis.sh
+  # and ensure they're set to the values we expect.
+  echo "$test: Checking specific settings in $settings_file"
+  testSetting $test $settings_file '=' INACTIVE 30
```

Comment:
Optional Suggestion: could we reorder the last three arguments to 'setting' 'separator' 'expected value', so that it reads from left to right?

Reply:
I change the function in general, but I think I have the ordering better now.